src/app/auth/resetPassword/index.tsx

```diff
@@ -0,0 +1,177 @@
+import { Link, router } from 'expo-router';
+import React, { useEffect, useRef, useState } from 'react';
+import { Alert, Text, View } from 'react-native';
+import { Icon as RNEIcon } from 'react-native-elements';
+
+import styles from './styles';
+import Icon from '../../../../assets/icons';
+import StyledButton from '../../../components/StyledButton/StyledButton';
+import UserStringInput from '../../../components/UserStringInput/UserStringInput';
+import colors from '../../../styles/colors';
+import globalStyles from '../../../styles/globalStyles';
+import { useSession } from '../../../utils/AuthContext';
+import PasswordComplexityText from '../../../components/PasswordComplexityText/PasswordComplexityText';
+import supabase from '../../../utils/supabase';
+
+function ResetPasswordScreen() {
+  const { updateUser, signOut } = useSession();
+  const [password, setPassword] = useState('');
+  const [passwordTextHidden, setPasswordTextHidden] = useState(true);
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [confirmPasswordTextHidden, setConfirmPasswordTextHidden] =
+    useState(true);
+  const [loading, setLoading] = useState(false);
+
+  const [hasUppercase, setHasUppercase] = useState(false);
+  const [hasLowercase, setHasLowercase] = useState(false);
+  const [hasNumber, setHasNumber] = useState(false);
+  const [hasLength, setHasLength] = useState(false);
+  const [isDifferent, setIsDifferent] = useState(false);
+  const [isMatching, setIsMatching] = useState(false);
+
+  const checkPassword = (text: string) => {
+    setPassword(text);
+    if (text !== '') {
+      setHasUppercase(text !== text.toLowerCase());
+      setHasLowercase(text !== text.toUpperCase());
+      setHasNumber(/[0-9]/.test(text));
+      setHasLength(text.length >= 8);
+      //need to check that it is different from old password
+    }
+  };
+
+  const checkPasswordMatchConfirmPassword = (text: string) => {
```

Comment:
Maybe rename this to `checkPasswordMatchesConfirmPassword` or something else? Up to what you think is more descriptive.